core/src/data/value/error.rs

```diff
@@ -95,6 +102,9 @@ pub enum ValueError {
     #[error("literal cast failed to date: {0}")]
     LiteralCastToDateFailed(String),
 
+    #[error("literal cast from {1} to {0} failed")]
+    LiteralCastToDataTypeFailed(DataType, String),
```

Comment:
This, too.
No use case in this PR. `LiteralCastToDataTypeFailed`

Reply:
Same here..